### ISSUE TITLE
cross-chain peg; e.g. to Agoric

### ISSUE THREAD

**Issue by dckc** [rchain-community/rstake#12]
I recently saw a demo of [Pegasus](https://www.peg-as.us/), a dApp for pegging ERTP rights to or from remote assets via the [Agoric Network API](https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/docs/networking.md).

It looks like one could implement that network API using the RChain web api and the [pegasus.js](https://github.com/Agoric/dapp-pegasus/blob/master/contract/src/pegasus.js) contract would largely just work. (Some refactoring is needed first: https://github.com/Agoric/dapp-pegasus/issues/2)

**Comment by michaelfig** [comment]
> It looks like one could implement that network API using the RChain web api and the [pegasus.js](https://github.com/Agoric/dapp-pegasus/blob/master/contract/src/pegasus.js) contract would largely just work.

Yep, that's the idea.  The main invariant is that each unique "network" connection has to be identified so that the protocol can decide on a unique name for the pegged assets.

**Comment by dckc** [comment]
Another possibility...

> Compared to IBC and PoS light client based cross-chain communication, Optics has weaker security guarantees, and a longer latency period. However, Optics may be implemented on any smart contract chain, with no bespoke light client engineering. Because it does not run a light client, Optics does not spend extra gas verifying remote chain block headers.
-- https://github.com/celo-org/optics-monorepo

It's not obvious how to do token transfers over Optics, actually...

cc @tgrospic @jimscarver
